src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -400,234 +443,223 @@ End Class
+        {{
+            await {0};
+        }}
+    }}
+}}
+            ";
+            return GetCSReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 11, 19, 11, endColumn);
         }
 
         #endregion
 
         #region VB - Diagnostic - Analyzer
 
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_ByteArray()
-        {
-            return AnalyzeVBAsync(@"
+        [Theory]
+        [InlineData("buffer, 0, s.Length",
+                    "buffer.AsMemory(0, s.Length)",
+                    false, 51)]
+        [InlineData("buffer, 0, s.Length",
+                    "buffer.AsMemory(0, s.Length)",
+                    true, 51)]
+        [InlineData("buffer, 0, s.Length, New CancellationToken()",
+                    "buffer.AsMemory(0, s.Length), New CancellationToken()",
+                    false, 76)]
+        [InlineData("buffer, 0, s.Length, New CancellationToken()",
+                    "buffer.AsMemory(0, s.Length), New CancellationToken()",
+                    true, 76)]
+        public Task VB_Analyzer_Diagnostic_VarByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
+        {
+            string source = @"
 Imports System
 Imports System.IO
 Imports System.Threading
-Class C
+Public Module C
     Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
-            Await fs.ReadAsync(buffer, 0, fs.Length)
+        Using s As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Dim buffer As Byte() = New Byte(s.Length - 1) {{}}
+            Await {0}
         End Using
     End Sub
-End Class
-            ", GetVBResult(9, 19, 9, 53));
-        }
-
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_Inline()
-        {
-            return AnalyzeVBAsync(@"
+End Module
+            ";
+            return GetVBReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 9, 19, 9, endColumn);
+        }
+
+        [Theory]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length)",
+                    false, 70)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length)",
+                    true, 70)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length, New CancellationToken()",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length), New CancellationToken()",
+                    false, 95)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length, New CancellationToken()",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length), New CancellationToken()",
+                    true, 95)]
+        public Task VB_Analyzer_Diagnostic_InlineByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
```

Comment:
Don't like hard coding the `endColumn` number.

Reply:
Can we change the code fix tests to be markup syntax based so there is no need for any magic numbers in tests?

Reply:
I cannot not use markup syntax `[||]` or `{||}`. I get an error about multiple diagnostics being detected. 
This was a suggestion given in the Analyzer PR as well, and we opted to use column names instead.

Reply:
Ah yes, sorry my bad that I forgot again. Ignore the suggestion about markup syntax.

Reply:
😃